src/classes/frDonation.cls

```diff
@@ -53,84 +54,139 @@ public class frDonation extends frModel implements frSyncable {
         return mappings;
     }
     
-    private Opportunity o;
+    private SObject o;
     
     public frDonation(Sync_Attempt__c syncRecord){
         super(syncRecord);
     }
-    
+
     public Boolean sync() {
+        //create Sobject
         Boolean result = false;
         Map<String, Object> request = getRequestBody();
-        o = new Opportunity();
+        String frId = getFunraiseId();
+
+        if(!frUtil.hasNPCobjects()) {
+            o = new Opportunity();
+            setOpportunityStage((Opportunity) o, String.valueOf(request.get('status')));
+        } 
+        else {
+            List<SObject> gtList = Database.query('SELECT Id, Name, DonorId, AcknowledgementDate FROM GiftTransaction WHERE fr_ID__c = :frId');
+            if(gtList.isEmpty()) {
+                o = Schema.getGlobalDescribe().get('gifttransaction').newSObject();
+            } 
+            else {
+                o = gtList.get(0);
+            }
+        }
+
         applyMappings(o, request);
-        setOpportunityStage(o, String.valueOf(request.get('status')));
-        
         String funraiseId = getFunraiseId();       
-        String supporterFunraiseId = String.valueOf(request.get('donorId'));        
+        String supporterFunraiseId = String.valueOf(request.get('donorId')); 
+        String supporterObjName = frUtil.hasNPCobjects() ? 'Account' : 'Contact';
+        String subsciptionObjName = frUtil.hasNPCobjects() ? 'GiftCommitment' : 'Subscription__c';
```

Comment:
I could use the .Type in replacement for the frUtil.hasNPCobjects(), but since its not the object API name i still need to use this kind of statement.